Makefile.features

```diff
@@ -1,3 +1,33 @@
 # Process FEATURES variables
 
 include $(RIOTBOARD)/$(BOARD)/Makefile.features
+
+
+# Resolve FEATURES_ variables
+# Their value will only be complete after resolving dependencies
+
+# Features that are required by the application but not provided by the BSP
+# Having features missing may case the build to fail.
+FEATURES_MISSING = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_REQUIRED)))
+
+# Features that are only optional and not required at the same time.
+# The policy is to use by default whenever they are provided by the BSP.
+FEATURES_OPTIONAL_ONLY = $(sort $(filter-out $(FEATURES_REQUIRED),$(FEATURES_OPTIONAL)))
+FEATURES_OPTIONAL_USED = $(sort $(filter $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
+# Optional features that will not be used because they are unavailable
+FEATURES_OPTIONAL_UNUSED = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
+
+# Features that are used for an application
+FEATURES_USED = $(sort $(FEATURES_REQUIRED) $(FEATURES_OPTIONAL_USED))
+
+# Used features that conflict when used together
+FEATURES_CONFLICTING = $(sort $(foreach conflict,$(FEATURES_CONFLICT),$(call _features_conflicting,$(conflict))))
```

Comment:
Not sure why but this is not working for me. I'm doing

`make -C tests/periph_rtt/ BOARDS=slstk3401a info-build`

while also setting `FEATURES_REQUIRED+=periph_rtc` and printing `FEATURES_CONFLICTING` right after the assignment, its always empty while `FEATURES_USED` contains `periph_rtt` and `periph_rtc` and `FEATURES_CONFLICT=periph_rtt:perih_rtc`. Am I missing something?

Reply:
The command is `BOARD=slstk3401a make -C tests/periph_rtt/ info-build`

But indeed the `CONFLICTING` seams to be not working, I will check what I am missing.

Reply:
Found it, the issue is in `tests/periph_rtt` that does `FEATURES_REQUIRED =` so ignored what you say in the environment.


This makes it work:

```diff
diff --git a/tests/periph_rtt/Makefile b/tests/periph_rtt/Makefile
index 6aabe4181..5c1392152 100644
--- a/tests/periph_rtt/Makefile
+++ b/tests/periph_rtt/Makefile
@@ -1,7 +1,7 @@
 BOARD ?= samr21-xpro
 include ../Makefile.tests_common

-FEATURES_REQUIRED = periph_rtt
+FEATURES_REQUIRED += periph_rtt

 DISABLE_MODULE += auto_init

```

```
FEATURES_REQUIRED+=periph_rtc BOARD=slstk3401a make -C tests/periph_rtt/ info-build
...
FEATURES_CONFLICT:     periph_rtc:periph_rtt
FEATURES_CONFLICT_MSG: "On the EFM32, the RTC and RTT map to the same hardware peripheral."
FEATURES_CONFLICTING:
         periph_rtc periph_rtt
```

Reply:
Weird, I could Have sworn I had changed `FEATURES_REQUIRED = periph_rtt`
to `FEATURES_REQUIRED += periph_rtt`... they were both showing up on FEATURES_REQUIRED and FEATURES_PROVIDED. I'll have to re-check tomorrow on my primary-system but it is working now.

Reply:
Found the typo in the command I was running